Lcass chem

lcass opened the pull request:
This is a complete update to chemistry , a few things need to be listed out.
1: This will not mess with trek chem , this stays as is.
2: The chemicals can be dangerous so admins may want to look out specifically in the new area.
3: This chemical update is used for the science department.

So what is added?
Lots of new machines , three which are not dangerous and 2 which can kill making the more dangerous chemicals much harder to produce. Lots of new medical chems , some which will be sought after for curing viruses and some that will reinstate the science medical links. Medical isn't as fun as it could be, because of this lots of new chemicals that are dangerous are added. This should spice up some traitor rounds , for instance the chemical emitrium causes people to shoot out emitter beams, great for destroying a crowd! An entire new department area is added , the chemical laboratory at the south of science includes all the machines in testing areas and is coated in fire locks to prevent spreading. No new jobs are added so lots of people can join in. Have fun with this.

lcass committed:
Chem holder for chem update

lcass committed:
chem machines for chem update

lcass committed:
procs for chem update

lcass committed:
chems for chem update

lcass committed:
meds for chem update

lcass committed:
meth for chem update

lcass committed:
recipes for chem update

lcass committed:
reagent containers for chem update

lcass committed:
areas for chem update

lcass committed:
construction for chem udpate

lcass committed:
machine circuits for chem update

lcass committed:
icons for chem update

lcass committed:
noob mistake fix

lcass committed:
chem map for chem update

Iamgoofball commented on code/modules/reagents/Chemistry-Holder.dm:
```diff
@@ -10,8 +10,10 @@ datum/reagents
+	var/list/present_machines = list(-1,-1,-1,-1,-1)
 
 datum/reagents/New(maximum=100)
+	processing_objects.Add(src) //tada she now ticks
```
remember to remove this from the processing_objects list on deletion

lcass replied:
Ugh I had a fix for this where it would only do it for certain reagents.

Iamgoofball commented on code/modules/reagents/Chemistry-Holder.dm:
```diff
@@ -492,3 +534,19 @@ atom/proc/create_reagents(var/max_vol)
 	reagents.my_atom = src
+
+//also a convenience , used to create smoke which will be used a lot in these reactions
+datum/proc/create_smoke(var/datum/reagents/holder, var/created_volume)
```
make sure to sanity check that holder and created_volume exist just in case

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+		M.stat = 1
+		..()
+		return
+	else
```
This is going to revive someone and then instantly kill them. I'd make this be in the reaction_mob().

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+	color = "#C8A5DC" // rgb: 200, 165, 220
+
+datum/reagent/medicine/sodiumf/on_mob_life(var/mob/living/M as mob)
+	if(M.stat != DEAD)
```
These DEAD stat checks really aren't needed, it doesn't process if the guy is dead anyways.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+
+datum/reagent/medicine/sodiumf/on_mob_life(var/mob/living/M as mob)
+	if(M.stat != DEAD)
+		M.reagents.remove_all_type(/datum/reagent/toxin, 1*REM, 0, 1)
```
that's kind of overpowered, don't you think?

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+
+datum/reagent/medicine/life/on_mob_life(var/mob/living/M as mob)
+	if(M.stat != DEAD)
+		M.setToxLoss(0)//deals minor toxin damage  designed to be very potent
```
nice copypasta

Iamgoofball replied:
also why would I ever use any chem other than this for healing

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+		M.setToxLoss(0)//deals minor toxin damage  designed to be very potent
+		M.hallucination = 0 //just a weeny bit
+		M.adjustFireLoss(-10)
+		M.adjustBruteLoss(-10)
```
why would I ever use aluminiumf over this

why would I use any other chem than this for healing

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm:
```diff
@@ -236,6 +278,111 @@ datum/reagent/medicine/anti_toxin/on_mob_life(var/mob/living/M as mob)
+				D.cure()
+	..()
+	return
+datum/reagent/medicine/zed
```
this reagent literally does nothing

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/New-Reagents.dm:
```diff
@@ -97,6 +97,76 @@ datum/reagent/nicotine/on_mob_addicted(var/mob/living/M as mob)
+	id = "meth"
+	description = "Baby Blue."
+	color = "#3333FF" // rgb: 205,133,63
+	addiction_rate = 0.2 
```
this is the most ugly addiction system I've ever seen

move it to the goonchem addiction framework, it'll work better + be nicer to read

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+	description = "Chilly."
+	color = "#75AC53"
+datum/reagent/toxin/cryonium/on_mob_life(var/mob/living/M as mob)
+	M.bodytemperature -= 3//this is how you avoid super boom
```
the enviroment balances this out too fast

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+		holder.present_machines[1] -= (5- holder.present_machines[1])
+	holder.remove_reagent(src.id,5)
+datum/reagent/toxin/pyrosium
+	name = "Pyrosium"
```
if you're not gonna make it work like pyrosium don't name it pyrosium

it's just going to confuse the hell out of people coming in from other servers/leaving for other servers

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+	color = "#75AC53"
+datum/reagent/toxin/pyrosium/on_mob_life(var/mob/living/M as mob)
+	M.bodytemperature += 3
+	holder.remove_reagent(src.id,5,safety = 1)
```
just make 5 the metabolization rate

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+		M.emote(pick("cough","sneeze"))
+datum/reagent/toxin/smokep/on_fart(var/mob/living/M as mob)
+	M.visible_message("<span class='danger'>[M] farts out a noxious cloud!</span>")
+	create_smoke(src, 10)
```
this isn't a reagents holder you're feeding in, you're feeding in /datum/reagent

this will runtime badly

lcass replied:
didn't notice , thanks for pointing it out.

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+	description = "A fairly nasty chemical used to produce potent medicines"
+	color = "#A300B3"
+datum/reagent/toxin/carbonf/on_mob_life(var/mob/living/M as mob)
+	M.adjustToxLoss(1)
```
> fairly nasty
> basic antitoxin heals more than it does a tick

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+datum/reagent/toxin/spark/on_mob_life(var/mob/living/M as mob)
+	if(prob(10))
+		M.adjustFireLoss(3)//extremely weak damage
+		var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread
```
shoot people with this when the station is being plasma flooded for maximum !!FUN!!

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+datum/reagent/toxin/dizinc/on_mob_life(var/mob/living/M as mob)
+	M.adjust_fire_stacks(6)
+	M.IgniteMob()
+	M.bodytemperature += 15 //it's getting HOT HOT HOT dunununu
```
why would I use pyrosium over this

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+
+datum/reagent/toxin/radgoop/on_mob_life(var/mob/living/M as mob)
+	if(prob(30))
+		M.radiation ++
```
> putting a space here

muh OCD

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+datum/reagent/toxin/dizinc/reaction_mob(var/mob/living/M, var/method=TOUCH, var/volume)
+	M.adjust_fire_stacks(6)
+	M.IgniteMob()
+	holder.clear_reagents()
```
why would I use any anti-toxin over this if I have a spray bottle of water on hand

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+	id = "alco"
+	description = "I'm wasted man!"
+	color = "#CAD15A"
+datum/reagent/toxin/alco/on_mob_life(var/mob/living/carbon/M as mob)
```
did you really just copypaste booze code

and kept the shitty data var usage

Iamgoofball commented on code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm:
```diff
@@ -120,6 +121,527 @@ datum/reagent/toxin/slimejelly/on_mob_life(var/mob/living/M as mob)
+datum/reagent/toxin/tabun
+	name = "Tabun"
+	id = "tabun"
+	description = "Made by your friendly neighbourhood nazis!"
```
> includes tabun
> 
> no clf3

Jaygriff committed:
Changed a few wall types to reinforced

Was two or three walls that needed to be reinforced rather than regular.

lcass committed:
create smoke check

blocks runtimes from nub coding

lcass committed:
defib fix

Fixes a chemical that was buggy. Thanks to goof for pointing this out.

lcass committed:
Balances sodiumf

Makes sodium f more balanced

lcass committed:
Toxin fixes

Fixes multiple issues pointed out by the wonderful goof. Thanks.

lcass committed:
Recipe update

Fixes chemicals that were previously bugged and could not be made , sorts out some side reactions that were occuring between chems and makes certain chemicals use more suitable reagents.

lcass committed:
Chemistry holder fix

Makes it so that the bluespace recombob actually works.

lcass committed:
processing objects

removes reagents from processing on delete

Jaygriff committed:
Door perms fixed, sinks added, fixed showers

Jaygriff committed:
Added wall radio and fixed piping issue I made

lcass committed:
Recipe fixes

some recipes were BORKED also adds bearium

lcass committed:
Chem machine ui update

Empty machines can have their beakers ejected and the distiller tells you if it is heating.

lcass committed:
Defib and virogone fix

fixes defib and makes virogone not cure alien embryos or uncurable diseases.

lcass committed:
sanity check

updates the santiy check to be more sane.

lcass committed:
YADF

Yet another defib fix. Checks the person is actually a human so we can revive them.

lcass committed:
defib fix again

now it can compile...
